pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -389,18 +433,62 @@ func (a *Assignment) findFlavorForResourceGroup(
 			}
 		}
 
-		if representativeMode > bestAssignmentMode {
-			bestAssignment = assignments
-			bestAssignmentMode = representativeMode
-			if bestAssignmentMode == Fit {
-				// All the resources fit in the cohort, no need to check more flavors.
-				return bestAssignment, nil
+		if features.Enabled(features.FlavorFungibility) {
+			if !shouldTryNextFlavor(representativeMode, cq.FlavorFungibility, needsBorrowing) {
+				bestAssignment = assignments
+				bestAssignmentMode = representativeMode
+				break
+			}
+			if representativeMode > bestAssignmentMode {
+				bestAssignment = assignments
+				bestAssignmentMode = representativeMode
+			}
+		} else {
+			if representativeMode > bestAssignmentMode {
+				bestAssignment = assignments
+				bestAssignmentMode = representativeMode
+				if bestAssignmentMode == Fit {
+					// All the resources fit in the cohort, no need to check more flavors.
+					return bestAssignment, nil
+				}
 			}
 		}
 	}
+
+	for _, assignment := range bestAssignment {
+		// When the workload needborrowing and there is another workload in cohort doesn't
+		// need borrowing, the workload needborrowing will come again. In this case we should
+		// not skip the previous flavors.
+		if flavorIdx == len(rg.Flavors)-1 || assignment.Mode == Fit {
```

Comment:
I set `assignment.FlavorIdx = -1` when `assignment.Mode == Fit` so that any workload need borrowing or fit comes again, it will not skip any flavors. (`assignment.FlavorIdx = -1` means no flavor would be skipped.)

set workload.LastAssigment to nil when a workload is skipped in scheduler.go should do the same thing, I think.

Reply:
I prefer we do the changes in scheduler.go, just for separation of concerns (the flavor assigner shouldn't know about skips).

Reply:
Ok, I understand. Have moved it to scheduler.go